jedi/evaluate/iterable.py

```diff
@@ -443,37 +444,56 @@ def py__iter__types(evaluator, types, node=None):
     return unite(py__iter__(evaluator, types, node))
 
 
-def py__getitem__(evaluator, types, index, node):
+def py__getitem__(evaluator, types, trailer):
```

Comment:
Ok, I hope I still fully recall... I went through a number of iterations here (not all of them well-informed), and by now it might or might not be necessary to have the changed signature.

I wanted the raw trailer. Note that `node` in the original signature was always the closing `]` (I think). My first reason for that was that I wanted the index before it went through `create_index_types` --- I seem to recall that in the end this might not have been necessary.
The other reason was that I wanted to feed the index-node (not just the index) to `get_types_for _typing_module`. This was to have somewhere to get the parent context from, when doing the forward reference resolving. Since later I moved the forward reference resolving back to parsing the annotations, this is not necessary anymore. However it may become necessary again if we revert that code change, and resolve FRs in `get_types_for _typing_module` once more.